core/state/statedb.go

```diff
@@ -650,6 +669,20 @@ func (s *StateDB) getDeletedStateObject(addr common.Address) *StateObject {
 			}
 		}
 	}
+
+	// ErrSnapshotStale may occur due to diff layers in the update, so we should try again in noTrie mode.
+	if s.NoTrie() && err != nil && errors.Is(err, snapshot.ErrSnapshotStale) {
```

Comment:
for other errors, do we need to handle too? since trie is no enabled, we should not go further if there is an error

Reply:
since we can not regenerate snapshot if we don't have trie, should we add a doc for handling this issue

Reply:
I think if get other errors, we should keep the original logic and let the block re-import